fortnite_api/cosmetics/br.py

```diff
@@ -0,0 +1,209 @@
+from ..enums import CosmeticBrSearchTag
+from ..http import HTTPClientT
+from ..utils import parse_time
+from .common import Cosmetic, CosmeticImages, CosmeticRarity, CosmeticSeries, CosmeticType
+
+
+class CosmeticSet:
+    """Represents a set that a given cosmetic belongs to.
+
+    Attributes
+    ----------
+    value: :class:`str`
+        The value of the set, also known as the name.
+    text: :class:`str`
+        The display text of this set. In the form, ``"Part of the {{value}} set."``
+    backend_value: :class:`str`
+        The backend value of this set.
+    """
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
+        self.value: str = data['value']
+        self.text: str = data['text']
+        self.backend_value: str = data['backendValue']
+
+
+class CosmeticIntroduction:
+    """Holds some metadata about when a cosmetic was introduced.
+
+    Attributes
+    ----------
+    chapter: :class:`int`
+        The chapter the cosmetic was introduced in.
+    season: :class:`str`
+        The season the cosmetic was introduced in. Can be "OG" if the cosmetic was introduced before seasons.
+    text: :class:`str`
+        The display text of this introduction. In the form, ``"Introduced in Chapter {{chapter}}, Season {{season}}."``
+    backend_value: :class:`int`
+        The backend value of this introduction.
+    """
+
+    def __init__(self, *, data: Dict[str, Any]) -> None:
+        self.chapter: int = int(data['chapter'])
+        self.season: str = data['season']
+        self.text: str = data['text']
+        self.backend_value: int = data['backendValue']
+
+
+class CosmeticVariantOption(Generic[HTTPClientT]):
+    """Represents a variant option for a cosmetic.
+
+    Attributes
+    ----------
+    tag: :class:`str`
+        The tag of the variant option.
+    name: :class:`str`
+        The option's name.
+    image: :class:`Asset`
+        The image of the variant option.
+    """
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
+        self.tag: str = data['tag']
+        self.name: str = data['name']
+        self.image: Asset[HTTPClientT] = Asset[HTTPClientT](http=http, url=data['image'])
+
+
+class CosmeticVariant(Generic[HTTPClientT]):
+    """Represents a variant for a cosmetic.
+
+    Attributes
+    ----------
+    channel: :class:`str`
+        The channel of the variant.
+    type: :class:`str`
+        The type of the variant.
+    options: List[:class:`CosmeticVariantOption`]
+        The options for the variant, if any.
+    """
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
+        self.channel: str = data['channel']
+        self.type: str = data['type']  # TODO: Move to enum eventually.
+
+        self.options: List[CosmeticVariantOption[HTTPClientT]] = [
+            CosmeticVariantOption[HTTPClientT](data=option, http=http) for option in data['options']
+        ]
+
+
+class CosmeticBr(Cosmetic[HTTPClientT]):
```

Comment:
Not sure about the name. BrCosmetics sounds better in my opionion.

Reply:
I agree with you, but this was done for a very specific reason. Because of the 6 closely related types, every cosmetic related item has been prefixed with `CosmeticX` for IntelliSense optimizations. It'll also be easier for users to use documentation as well